lib/plugins/authad/auth.php

```diff
@@ -326,14 +331,127 @@ public function isCaseSensitive() {
     }
 
     /**
+     * @param array $filter
+     * @return string
+     */
+    protected function _constructSearchString($filter){
+        if (!$filter){
+            return '*';
+        }
+        $result = '*';
+        if (isset($filter['name'])) {
+            $result .= ')(displayname=*' . $filter['name'] . '*';
```

Comment:
I guess the filter needs escaping? Otherwise a parenthesis or asterisk might break the filter string. The adLDAP library has an escaping function for filter terms.